golang/pkg/crane/crux/deploy.go

```diff
@@ -16,36 +17,56 @@ import (
 	common "github.com/dyrector-io/dyrectorio/protobuf/go/common"
 )
 
-func GetDeployments(ctx context.Context, namespace string) []*common.ContainerStateItem {
+func WatchDeploymentsByPrefix(ctx context.Context, namespace string) (*grpc.ContainerWatchContext, error) {
 	cfg := grpc.GetConfigFromContext(ctx).(*config.Configuration)
 	client := k8s.NewClient(cfg)
 
 	deploymentHandler := k8s.NewDeployment(ctx, cfg)
-	deployments, err := deploymentHandler.GetDeployments(ctx, namespace, cfg)
-	if err != nil {
-		log.Error().Err(err).Stack().Send()
-		return nil
-	}
-
 	svcHandler := k8s.NewService(ctx, client)
-	svc, err := svcHandler.GetServices(namespace)
-	if err != nil {
-		log.Error().Err(err).Stack().Send()
-	}
 
-	podsByDeployment := make(map[string][]corev1.Pod)
-	for i := 0; i < len(deployments.Items); i++ {
-		deployment := deployments.Items[i]
-		pods, err := deploymentHandler.GetPods(namespace, deployment.Name)
-		if err != nil {
-			log.Error().Err(err).Stack().Send()
-			return nil
-		}
+	eventChannel := make(chan []*common.ContainerStateItem)
 
-		podsByDeployment[deployment.Name] = pods
-	}
+	// TODO: Proper kubernetes event handling
```

Comment:
Let's check if we can use kube's event api with the current .proto changes in the future.